torch/distributed/device_mesh.py

```diff
@@ -652,6 +690,18 @@ def get_coordinate(self) -> Optional[List[int]]:
             """
             return self._coordinate_on_dim if self._coordinate_on_dim else None
 
+        def _flatten(self) -> "DeviceMesh":
```

Comment:
why can't we make this a public API?

Reply:
I was thinking let Chien-chin try it out first to see if there is any minor UX improvement we want to make before making it public. Wdyt?

Reply:
sure we can make it public later, I don't think it's quite a concern though given that we are working on the main branch which is separate from the release schedule, but your call